src/core/service.c

```diff
@@ -2659,6 +2662,67 @@ static void service_enter_reload_by_notify(Service *s) {
                 log_unit_warning(UNIT(s), "Failed to schedule propagation of reload, ignoring: %s", bus_error_message(&error, r));
 }
 
+static bool service_should_reload_extensions(Service *s) {
+        int r;
+
+        assert(s);
+
+        /* Only support this for notify-reload service types. */
+        if (s->type != SERVICE_NOTIFY_RELOAD)
+                return false;
```

Comment:
i still think this condition is just wrong and should not exist. not sure why system-wide confext handling should be treated differently than per-service confext handling

Reply:
https://github.com/systemd/systemd/pull/33995/files#r1757258832

Reply:
yeah, everyone disagrees with your take on that, @bluca

Reply:
That's fine, anybody who disagrees can spend their own time to implement any additional feature they like on top of this one. This is not a veto on adding new features, it's a rejection of scope creep and of the attempt to impose additional, unrelated, unknown, time-consuming and difficult work on _us_. Anybody else can work on it later if they like, no problem at all from me.

Reply:
frankly, i don't think i should have bdfl powers. But then again I don't get the sense the artificial restriction has many friends besides @bluca himself.

i find it utterly confusing to bind this to Type=reload-notify. I see zero connection to confext from that.

Reply:
> This is certainly a valid stance to have but when it comes down to it it's also totally valid for @poettering to not give this the seal of approval if this isn't addressed in this PR.

Scope creep is a huge problem in this project. Someone comes with solution X for problem A -> "while there can you also please implement Y, W, Z to solve B, C, D". We are all guilty of it at some point including myself, but we _really_ need to make a collecting effort to stop it, as it just drives people away. So I don't think it should be a valid stance to say "you can't have feature X until you implement feature Y".

> I think the most productive way to resolve this discussion is to write a detailed comment explaining why allowing this for arbitrary services is hard and to give multiple concrete examples of the issues with it to hopefully convince @poettering that this is full of footguns.

I think I've already done that plenty, but: this is a separate feature, with a separate lifecycle requirement and handling, separate use cases, separate implementation, that we have _no concrete use case for_ which means we have no idea how to implement it as the semantics are utterly unclear, besides some wild and speculative guessing, that are unlikely to survive contact with the real world, since it involves completely random, unpredictable and custom user-provided executable(s) that we just have no clue what they do. Would this run before it? After it? During it? Would they have access to it at all? No idea, there's no use case for it, so we are making it up as we go. This is especially important once this is updated to be handled asynchronously, as the state machine handling would be WILDLY different, and require a metric ton of extra, complex and risky work. Note again that I am not vetoing anybody from doing such work if they want to (on the contrary, feel free to, at any point), but also I am not willing to do it myself or to commit any resources from our team.

> If that doesn't work then I'm afraid your only option is to implement it.

There's also the option that this feature just doesn't go upstream and our investment into this project's upstream for immutable and integrity-verified configuration handling grinds to a halt, and instead we start moving to either a downstream private (or public) fork, or even to alternative tools to handle workloads. It's not a great option, but it is an option.